src/Raven.Server/Documents/Sharding/ShardedDocumentDatabase.cs

```diff
@@ -77,6 +89,41 @@ protected override void OnDatabaseRecordChanged(DatabaseRecord record)
         }
     }
 
+    private ConcurrentDictionary<long, Task> _confirmations = new ConcurrentDictionary<long, Task>();
+
+    public void TryConfirmMigration()
```

Comment:
This is where we do the immediate ack? Triggered by the db record bucket changing?